services/provider/server/server.go

```diff
@@ -1079,3 +1083,51 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
 
 	return &pb.PeerStorageClusterResponse{}, nil
 }
+
+func (s *OCSProviderServer) RequestMaintenanceMode(ctx context.Context, req *pb.RequestMaintenanceModeRequest) (*pb.RequestMaintenanceModeResponse, error) {
+	// Get storage consumer resource using UUID
+	consumerObj, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		klog.Error(err)
+		return nil, status.Errorf(codes.NotFound, "Failed to get storageConsumer resource")
+	}
+
+	_, hasAnnotation := consumerObj.GetAnnotations()[util.RequestMaintenanceModeAnnotation]
+	if req.Enable != hasAnnotation {
+		if req.Enable {
+			util.AddAnnotation(consumerObj, util.RequestMaintenanceModeAnnotation, "")
+		} else {
+			delete(consumerObj.GetAnnotations(), util.RequestMaintenanceModeAnnotation)
+		}
+
+		err = s.consumerManager.Update(ctx, consumerObj)
+		if err != nil {
+			klog.Error(err)
+			return nil, status.Errorf(codes.Internal, "Failed to update storageConsumer resource")
+		}
+	}
+
+	return &pb.RequestMaintenanceModeResponse{}, nil
+}
+
+func (s *OCSProviderServer) isSystemInMaintenanceMode(ctx context.Context) bool {
+	// TODO: Refactor to use presence of cephRBDMirror CR
+	// Looking for the rbd mirror deployment until rook ensures that the deployment is deleted
```

Comment:
Opened an issue for this in Rook https://github.com/rook/rook/issues/15031

Reply:
Thanks